spec/community.mediawiki

```diff
@@ -55,3 +55,29 @@ of any unfilled orders.
 
 Less drastic punitive measures such as a cool-down period may be considered
 for minor, first-time or infrequent conduct violations.
+
+===Penalization Notification===
+
+In the event that a rule is broken, the client will be sent a notification that
+includes the broken rule, a detailed description of the offense, the time of
+the penalization, and the duration. If the client is offline, this information
+can be found in the [[comm.mediawiki/#Session_Authentication|connect response]].
+
+'''Notification route:''' <code>penalize</code>, '''originator: ''' dex
+
+<code>payload</code>
+{|
+! field      !! type   !! description
+|-
+| accountid   || string || hex encoding of the client's account ID
+|-
```

Comment:
A `connect`ed websocket connection is specific to an account, so the `accountid` field is superfluous.

Reply:
My thinking was that the signature would be meaningless after the fact unless the accountid was included. While you could prove that a ban was dealt out from a particular server, you couldn't prove to whom it was directed. Although for simply verifying the server and communicating, it is not needed. I will take it out since there may never be need for the ban receiver to prove they were banned.

Reply:
I guess someone could reuse one of these messages though, if it didn't have the to whom. The signature will check out, only the timestamp will be old. Can that in any way be used as an attack vector? OK because wss is safe?

Reply:
> My thinking was that the signature would be meaningless after the fact unless the accountid was included. 

That's a good point. We can leave it for now and decide on later updates what's needed for the new system.